### PR TITLE
Add cred checking docs and config file

### PR DESCRIPTION
**Pull request by hillaryj**
#  #

## 🗣 Description ##

Adds a how-to guide for checking a repository for credentials, PII, and other things that would be bad to commit to a public repository.

## 💭 Motivation and Context ##

On our team, we generally use this tool on-demand rather than as a pre-commit hook as [cloud.gov does with its `caulking` repository](https://github.com/cloud-gov/caulking).

After running this tool, a person should review the results, which contain a number of false positives like referencing `username` variables in python code due to broad search patterns defined in `gitleak.toml`. Then, any actionable results can be remediated and a path to bringing a repository public can be determined.

## 🧪 Testing ##

I've run this tool several times and this documentation and rules file will allow others to do so as well.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] _All_ future TODOs are captured in issues, which are referenced
      in code comments.
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [x] Tests have been added and/or modified to cover the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

File: config/gitleaks-guide.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+finished installation, you can use the `--repo-path=<relative path>` option.
+
+Useful flags:
+
+- `--config=<file-path>` specifies the configuration file to use
+- `--repo-path=<relative-path>` specifies a local repository folder to check
+- `--verbose` gives you more details about each item found
+- `--pretty` formats the output so it's more human-readable
+
+```console
```
Comment (jsf9k) [inline config/gitleaks-guide.md]:
```suggestion
```sh
```

File: config/gitleaks-guide.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+such as:
+
+```console
+WARN[2020-12-08T10:47:03-05:00] 7 leaks detected. 24 commits scanned in 325 milliseconds 936 microseconds
+```
+
+If you want to save the console output to file so you can review it at your
+leisure, use the `>` syntax and specify a location and filename.
+
+```console
```
Comment (jsf9k) [inline config/gitleaks-guide.md]:
```suggestion
```sh
```

**Review by jsf9k — APPROVED** [review]
I left two minor comments for your suggestion.  Thanks for writing this up!

File: config/gitleaks-guide.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+by 18F's cloud.gov.
+
+Please note that our [gitleaks configuration file](gitleaks.toml) results in a
+number of false positives for things like code referencing "user_name" etc.
+
+Results should be reviewed by humans.
+
+## Installation ##
+
+This assumes you are on MacOS with HomeBrew installed.
```
Comment (dav3r) [inline config/gitleaks-guide.md]:
```suggestion
This assumes you are on MacOS with [HomeBrew](https://brew.sh) installed.
```

**Review by dav3r — APPROVED** [review]
Very nice- thanks @hillaryj!  🔭 

I had only one minor suggestion to add a link to some text.

**Pull request merged**